[aot_inductor] Fix issues in pre_grad passes

frank-wei opened the pull request:
Summary:

Fixed a bug in `sink_cat_after_pointwise` pass for PT IR. The root cause is asumption of existence of input in kwargs or args

 
Differential Revision: D55617545




cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler @amjames @desertfire @chauhang

hl475 reviewed (approved):
Is OSS failure relevant?

aakhundov commented on torch/_inductor/fx_passes/pre_grad.py:
```diff
@@ -478,8 +478,9 @@ def cat_args(tensors, dim=0):
                     return tensors, dim
 
                 tensors, dim = cat_args(*node.args, **node.kwargs)
+                new_kwargs = {x: user.kwargs[x] for x in user.kwargs if x != "input"}
```
```suggestion
                new_kwargs = {name: val for name, val in user.kwargs.items() if name != "input"}
```

frank-wei committed:
[aot_inductor] Fix issues in mts_gpu_benchmark and pre_grad passes to support AF_OC gated module (#123181)

Summary:

This diff helps to support request from https://fb.workplace.com/groups/gpuinference/permalink/2767700120045305/
Fixed a few things:
- found an issue in mts_gpu_benchmark, we cast the model to float at the beginning of the program. The fix was to fix issues for bf16 lowering since we encountered some models which are casted to fp16 and we need to cast to fp32 first.
The solution is to disable the casting and ask the user to set dense_type as fp32 at config level
- fixed a bug in `sink_cat_after_pointwise` pass for PT IR. The root cause is asumption of existence of input in kwargs or args

Reviewed By: hl475

Differential Revision: D55617545

clee2000 commented:
@pytorchbot merge -f "merged internally"
